src/benchmark_runner.cc

```diff
@@ -155,6 +164,10 @@ class BenchmarkRunner {
            internal::ARM_DisplayReportAggregatesOnly);
       run_results.file_report_aggregates_only =
           (b.aggregation_report_mode & internal::ARM_FileReportAggregatesOnly);
+      if (b.threads > 1 && perf_counters_measurement.IsValid()) {
```

Comment:
given you're checking this here, maybe you don't need the check above in the initializer list and can avoid storing the `perf_counters_measurement_ptr`?

Reply:
there are a few places we need to pass the counters pointer, `perf_counters_measurement_ptr` simplifies needing to check in every place whether we can do that.